File: android/src/main/kotlin/com/tsitser/background_http_plugin/domain/entity/HttpRequest.kt
package com.tsitser.background_http_plugin.domain.entity

/**
 * HTTP request model
 */
data class HttpRequest(
    val url: String,
    val method: String,
    val headers: Map<String, String>? = null,
    val body: String? = null,
    val queryParameters: Map<String, String>? = null,
    val timeout: Int? = null,
    val multipartFields: Map<String, String>? = null,
    val multipartFiles: Map<String, MultipartFile>? = null,
    val requestId: String? = null,
    val retries: Int? = null,
    val stuckTimeoutBuffer: Int? = null,
    val queueTimeout: Int? = null
)

/**
 * Multipart file model
 */
data class MultipartFile(
    val filePath: String,
    val filename: String? = null,
    val contentType: String? = null
)

File: ios/Classes/Domain/Entity/HttpRequest.swift
import Foundation

/// HTTP request model
struct HttpRequest {
    let url: String
    let method: String
    let headers: [String: String]?
    let body: String?
    let queryParameters: [String: String]?
    let timeout: Int?
    let multipartFields: [String: String]?
    let multipartFiles: [String: MultipartFile]?
    let requestId: String?
    let retries: Int?
    let stuckTimeoutBuffer: Int?
    let queueTimeout: Int?
}

/// Multipart file model
struct MultipartFile {
    let filePath: String
    let filename: String?
    let contentType: String?
}

